float64[] params
float64 t
float64 dT
